From ba99371da0e38f7eaeed8fc9a8892f6636e27e1c Mon Sep 17 00:00:00 2001
From: yeahme49 <58564160+yeahme49@users.noreply.github.com>
Date: Mon, 23 Mar 2020 11:32:43 -0500
Subject: [PATCH] open python file in binary mode

On Windows, opening the script in text mode can cause issues when scripts having linux line endings. Opening in binary mode is preferred.

ref: https://bugs.python.org/issue32809
ref: https://github.com/python/cpython/pull/12647/files
---
 xbmc/interfaces/python/PythonInvoker.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/interfaces/python/PythonInvoker.cpp b/xbmc/interfaces/python/PythonInvoker.cpp
index 62a30be9a6f0..ca96294d1fa5 100644
--- a/xbmc/interfaces/python/PythonInvoker.cpp
+++ b/xbmc/interfaces/python/PythonInvoker.cpp
@@ -322,7 +322,7 @@ bool CPythonInvoker::execute(const std::string& script, const std::vector<std::w
         return false;
       }
 #endif
-      FILE* fp = _Py_fopen(nativeFilename.c_str(), "r");
+      FILE* fp = _Py_fopen(nativeFilename.c_str(), "rb");
 
       if (fp != NULL)
       {
